src/sdk/bucketspace_fs.js

```diff
@@ -353,13 +353,27 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
                 if (!namespace_bucket_config) throw new RpcError('INTERNAL_ERROR', 'Invalid Bucket configuration');
 
                 if (namespace_bucket_config.should_create_underlying_storage) {
-                    // 1. delete underlying storage
+                    // 1. delete underlying storage (ULS = Underline Storage)
                     dbg.log1('BucketSpaceFS.delete_bucket: deleting uls', this.fs_root, namespace_bucket_config.write_resource.path);
                     const bucket_storage_path = path.join(this.fs_root, namespace_bucket_config.write_resource.path); // includes write_resource.path + bucket name (s3 flow)
-                    await ns.delete_uls({ name, full_path: bucket_storage_path }, object_sdk);
+                    try {
+                        await ns.delete_uls({ name, full_path: bucket_storage_path }, object_sdk);
+                    } catch (err) {
+                        dbg.warn('delete_bucket: bucket name', name, 'with bucket_storage_path', bucket_storage_path,
+                            'got an error while trying to delete_uls', err);
+                        // in case the ULS was deleted - we will continue
+                        if (err.rpc_code !== 'NO_SUCH_BUCKET') throw err;
+                    }
                 } else {
                     // 2. delete only bucket tmpdir
-                    const list = await ns.list_objects({ ...params, bucket: name, limit: 1 }, object_sdk);
+                    let list;
+                    try {
+                        list = await ns.list_objects({ ...params, bucket: name, limit: 1 }, object_sdk);
+                    } catch (err) {
```

Comment:
I suggest moving the check to the outer catch instead of having it twice here

Reply:
If I would remove this internal try-catch I would skip the next step which is removing the config file, and I don't want that to happen.